submissions/Ant-C-tech/memory-pair-game/js/app.js

```diff
@@ -0,0 +1,278 @@
+    while (TASK.length < CARDS_NUMBER / 2) {
+        const currentItem = hiragana[_getRandomIntInclusive(0, hiragana.length - 1)]
+        let flag = false
+        for (const item of TASK) {
+            if (item.jap === currentItem.jap) {
+                flag = true
+            }
+        }
+        if (flag === false) {
+            TASK.push(currentItem)
+        }
+    }
+    TASK = TASK.concat(TASK)
+}
+
+function createGreeting() {
+    const greeting = document.createElement('h3')
+    greeting.textContent = 'Kottans present'
+    greeting.classList.add('greeting')
+
+    greeting.addEventListener('click', function() {
+        GAMEPLAY_AUDIO.play()
+        changeContent(createStartGameScr(), showHeader)
+    }, { once: true })
+
+    return greeting
+}
+
+function createStartGameScr() {
+    const startBtnWrapper = new DocumentFragment()
+
+    const startBtnIco = document.createElement('img')
+    startBtnIco.setAttribute('src', 'img/lucky-cat-toy-svgrepo-com.svg')
+    startBtnIco.setAttribute('alt', 'mentor')
+    startBtnIco.classList.add('startBtnIco')
+
+    const startBtn = document.createElement('h3')
+    startBtn.textContent = 'Start Game'
+    startBtn.classList.add('startBtn')
+
+    startBtnWrapper.append(startBtnIco, startBtn)
+    startBtn.addEventListener('click', () => {
+        changeContent(createGameField(), phoneAdaptation)
+        hideElem(DESCRIPT, DESCRIPT_HIDE_ANIMATION, false)
+        window.addEventListener('resize', phoneAdaptation)
+    }, { once: true })
+
+    return startBtnWrapper
+}
+
+function createGameField() {
+    const gameField = document.createElement('div')
+    gameField.classList.add('cardsContainer')
+
+    for (let index = 0; index < CARDS_NUMBER; index++) {
+        const cardContent = TASK.splice(_getRandomIntInclusive(0, TASK.length - 1), 1)
+
+        const card = document.createElement('div')
+        card.className = `card ${ANIMATE_CSS_CLASS}`
+        card.setAttribute('data-value', cardContent[0].jap)
+
+        const cardBack = document.createElement('img')
+        cardBack.setAttribute('src', 'img/card-back2.jpg')
+        cardBack.setAttribute('alt', 'japanese ornament')
+        cardBack.classList.add('cardBack')
+
+        const cardFace = document.createElement('div')
+        cardFace.classList.add('cardFace')
+        const japText = document.createElement('p')
+        japText.innerText = cardContent[0].jap
+        japText.classList.add('japText')
+        const engText = document.createElement('p')
+        engText.innerText = cardContent[0].eng
+        engText.classList.add('engText')
+        cardFace.append(japText, engText)
+        card.append(cardBack, cardFace)
+
+        card.addEventListener('click', () => {
+            card.classList.add('main-notActive')
+            card.classList.add('card-rotate')
+            cardBack.classList.add('cardBack-rotate')
+            cardBack.addEventListener('transitionend', function() {
```

Comment:
Use event delegation here as well. It'll help you to increase the performance of your app.
More event listeners on the elements - less performance your app has